nexus_constructor/json/load_from_json.py

```diff
@@ -231,76 +231,85 @@ def _replace_placeholder(self, placeholder: str):
             }
         return None
 
-    def _read_json_object(self, json_object: Dict, parent_node: Group = None):
+    def _read_json_object(
+        self, json_object: Optional[Dict], parent_node: Optional[Group] = None
+    ):
         """
         Tries to create a component based on the contents of the JSON file.
         :param json_object: A component from the JSON dictionary.
         :param parent_name: The name of the parent object. Used for warning messages if something goes wrong.
         """
-        nexus_object: Union[Group, FileWriterModule] = None
+        nexus_object: Union[Group, FileWriterModule, None] = None
         use_placeholder = False
         if isinstance(json_object, str) and json_object in PLACEHOLDER_WITH_NX_CLASSES:
             json_object = self._replace_placeholder(json_object)
             if not json_object:
-                return
+                return None
             use_placeholder = True
-        if (
-            CommonKeys.TYPE in json_object
-            and json_object[CommonKeys.TYPE] == NodeType.GROUP
-        ):
-            try:
-                name = json_object[CommonKeys.NAME]
-            except KeyError:
-                self._add_object_warning(CommonKeys.NAME, parent_node)
+        if json_object:
```

Comment:
This comment is about 247 (if json_object) rather than 248 I think; this is related to the PR as it suppresses an error where json_object does not exist.  In such a scenario the effect is the same (no logic would run) but this makes it more explicit, both for Pylance but also for any reviewers of the code.  This comment did raise the issue however that there was no else condition (that runs if no applicable json_objects occur) so a warning has been added!